debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/rest/DebeziumMongoDbConnectorResource.java

```diff
@@ -41,6 +59,48 @@ public Connector getConnector() {
         return new MongoDbConnector();
     }
 
+    protected MongoDbConnection primary(MongoDbTaskContext context) throws DebeziumException {
+        ReplicaSet replicaSet = new ReplicaSet(context.getConnectionContext().connectionString());
+        return context.getConnectionContext().connect(replicaSet, context.filters(), (s, throwable) -> {
+            throw new DebeziumException(s, throwable);
+        });
```

Comment:
Note that this likely behaves differently than you would expect in case of sharded cluster and `connection.mode=replica_set` as it creates a connection from the intial connection string -- a different approach than this connection mode utilises later in the connector. 

Nevertheless, I think for the purpose of validation it should be fine. However, in such case I'd avoid using our internal MongoDbConnection and ReplicaSet classes and rather use the MongoDB client for this "seed connection" via `context.getConnectionContext().connect()`

Reply:
`MongoDbConnection` exists mainly for historical purposes -- it used to be much more complex. It's been gradually trimmed down with changes to overall connection management, error handling and so on. Next stage of trimming will be when we deprecate the replica_set connection mode

Reply:
Also note that `context.getConnectionContext().connectionString()` is a bit wonky as well as it returns the connection string as provided by the user which is not necessarily the exact cs used by the connector. That is the case when user doesn't specify the rs name via `replicaSet` parameter. In such case `ReplicaSetDiscovery#getReplicaSets()` emits a warning and tries to  determine the RS name form cluster description. 

Per se this is not a problem as Java's MongoDB driver can deal with this (however other drivers can't). 

The final connection string (or multiple in case of sharded cluster and `connection.mode=replica_set`) is then passed to the connector task via `mongodb.internal.task.connection.strings`. `MongoDbTask` then takes this value and reconstructs the internal `ReplicaSet` instances out of it.

Reply:
Although your approach will work I'm nut sure how problematic is the fact that connection logic of REST extension is different from what the connector does

Reply:
Same as above. It was just copied from the UI backend and I did not write the code there. I think it was maybe also only copied from legacy Mongo DB connector code.

Reply:
I'm sorry, but this shouldn't be used as an argument. The code wasn't frozen a year ago.

Using `context.getConnectionContext().connect()` should be better. However, we can definitively setup a call and explore the options. Feel free to ping me and we can look into it. We should probably discuss whether this validation makes sense for mongo anyway.

Reply:
Discussed after triaging call and continued on chat. Should be good now.